.github/actions/terraform-exec/action.yml

```
@@ -0,0 +1,96 @@
+name: Terraform execute
+description: Run `terraform` [command] with AWS credentials fetched through Vault.
+
+inputs:
+  vault-address:
+    description: Vault address
+    required: true
+  working-directory:
+    description: Directory to run the command in
+    required: false
+    default: .
+  command:
+    type: choice
+    description: Terraform command to run
+    required: true
+    options:
+      - plan
+      - apply
+  env:
+    description: Environment to run the command within
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - uses: hashicorp/setup-terraform@v2
+
+    - name: Authenticate Vault
+      id: secrets
+      uses: hashicorp/vault-action@v2
+      with:
+        exportToken: true
+        url: ${{ inputs.vault-address }}
+        method: jwt
+        role: ${{ inputs.env }}
+        # Even though it could look like separate calls to fetch the secrets
+        # the responses here are cached, so we're only issuing a single set of credentials
+        secrets: |
+          aws/creds/${{ inputs.env }}-deploy access_key | AWS_ACCESS_KEY_ID ;
+          aws/creds/${{ inputs.env }}-deploy secret_key | AWS_SECRET_ACCESS_KEY ;
+          aws/creds/${{ inputs.env }}-deploy security_token | AWS_SESSION_TOKEN
+
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v2
+      with:
+        aws-access-key-id: ${{ steps.secrets.outputs.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ steps.secrets.outputs.AWS_SECRET_ACCESS_KEY }}
+        aws-session-token: ${{ steps.secrets.outputs.AWS_SESSION_TOKEN }}
+        aws-region: us-east-1
+
+    - name: "[${{ inputs.env }}] Terraform Init"
```

Given that `plan` or `apply` is lowercase, I'd change this for consistency as well
```suggestion
    - name: "[${{ inputs.env }}] Terraform init"
```

---

The plan/apply will be lower-case with `${{ inputs.cmd }}` though

---

Yes, this is why I suggested to make this title lowercase as well 🙂

---

Oh sorry i saw the diff as the opposite. I'm blind xd

---

I wish I could say that never happens to me 😂